### PR TITLE
Add default leading and trailing trivia to tokens from `Format`

### PR DESCRIPTION
**Pull request by ahoppen**
This frees `TokenSyntax` from any implicit formatting and sinks all formatting logic into the `Format` type for a clearer separation of concerns.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/generate-swift-syntax-builder/Templates/FormatFile.swift [hunk 1]
```diff
@@ -225,3 +226,38 @@ private func createBuildableCollectionNodeFormatFunction(node: Node) -> Function
...
+        }
+      }
+      SwitchCase(label: SwitchCaseLabel(caseItems: CaseItem(pattern: ExpressionPattern(expression: MemberAccessExpr(name: "eof"))))) {
+        ReturnStmt(expression: "syntax")
+      }
+    }
+  }
+}
+
+func createWithLeadingWithTrailingTriviaCall(token: TokenSpec) -> CodeBlockItem {
```
Comment (fwcd) [inline Sources/generate-swift-syntax-builder/Templates/FormatFile.swift:250]: Can we make this private?

```suggestion
private func createWithLeadingWithTrailingTriviaCall(token: TokenSpec) -> CodeBlockItem
```

**Review by fwcd — APPROVED** [review]
Looks good, a minor stylistic note below.

The only thing that I am a bit concerned about is that this regresses generation time slightly (9 s on `main` vs 12 s on this branch, with a debug build on my machine), probably due to the additional `withXTrivia` calls. Since token generation is usually on the hot path and happens pretty often, even small changes can have an impact there.

In any case that is probably for another PR.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Pull request merged**